python/raft-dask/pyproject.toml

```diff
@@ -35,7 +35,6 @@ dependencies = [
     "distributed-ucxx==0.40.*,>=0.0.0a0",
     "joblib>=0.11",
     "numba>=0.57",
-    "numpy>=1.23,<2.0a0",
     "pylibraft==24.10.*,>=0.0.0a0",
```

Comment:
AFAICT `numpy` is not used in `raft-dask`

```
$ git grep numpy -- python/raft-dask/ | wc -l
       0
```

Also the Conda package doesn't include it

<details>

https://github.com/rapidsai/raft/blob/f1cc0fb79d2f40f6cc266c60928ed70ea64c843a/conda/recipes/raft-dask/meta.yaml#L61-L76

</details>

The origin of this `numpy` line in `raft-dask`'s wheels traces back to the addition of wheels in PR ( https://github.com/rapidsai/raft/pull/1013 ). Though not seeing any discussion about why `numpy` was added there. Even at the time conda `raft-dask` packages did not depend on `numpy`. Am guessing this was just copied into the different RAFT wheel packages. So it is likely just an oversight that this hasn't already been cleaned up

<hr>

Though `pylibraft`, used by `raft-dask`, does use `numpy`

```
$ git grep numpy -- python/pylibraft/ | wc -l
      52
```

Also `pylibraft` already includes `numpy` as a dependency

https://github.com/rapidsai/raft/blob/a773fa5e03eca840d9c92412a8ca1f2eeed4a455/conda/recipes/pylibraft/meta.yaml#L68

https://github.com/rapidsai/raft/blob/a773fa5e03eca840d9c92412a8ca1f2eeed4a455/python/pylibraft/pyproject.toml#L35

So we can be confident `numpy` will be in the environment `raft-dask` is installed into even though it appears `raft-dask` itself doesn't need it

<hr>

Given all of this, took this opportunity to clean this up and align our packages on the handling of this dependency